packages/core/src/rest/SponsorBlock.ts

```diff
@@ -0,0 +1,60 @@
+import { VideoID, Category, Segment } from './SponsorBlock.types';
+import _ from 'lodash';
+
+const BASE_URL = 'https://sponsor.ajay.app';
+export const ALL_CATEGORIES = ['sponsor', 'intro', 'outro', 'interaction', 'selfpromo', 'music_offtopic'];
+
+export async function getSegments (videoID: VideoID, categories?: Category[]): Promise<Segment[]> {
+  let query = `?videoID=${videoID}`;
+
+  if (!categories) {
+    query += `&categories=${JSON.stringify(ALL_CATEGORIES)}`;
+  } else if (categories.length > 1) {
+    query += `&categories=${JSON.stringify(categories)}`;
+  } else if (categories.length === 1) {
+    query += `&category=${categories[0]}`;
+  } 
+
+  return new Promise((resolve) => {
```

Comment:
This can be simplified like this:
```typescript
try {
    const result = await fetch(`${BASE_URL}/api/skipSegments${query}`);
    const json = await result.json();

    //rest of the processing
} catch(e) {
    // Log error
} finally {
    return [];
}
```